syntax = "proto3";

service ModelService {
  rpc TrainModel (TrainRequest) returns (TrainResponse);
  rpc Predict (PredictRequest) returns (PredictResponse);
  rpc DeleteModel (DeleteRequest) returns (DeleteResponse);
}

message TrainRequest {
  string model_type = 1;
  map<string, string> hyperparameters = 2;
  repeated float target = 3;
  repeated TrainData train_data = 4; // Define train_data as a list of lists
}

message TrainData {
  repeated float features = 1;
}

message TrainResponse {
  string model_id = 1;
}

message PredictRequest {
  string model_id = 1;
  repeated float data = 2;
}

message PredictResponse {
  repeated float predictions = 1;
}

message DeleteRequest {
  string model_id = 1;
}

message DeleteResponse {
  bool success = 1;
}
